akka-coordination/src/main/scala/akka/coordination/lease/scaladsl/Lease.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.coordination.lease.scaladsl
+
+import akka.annotation.ApiMayChange
+import akka.coordination.lease.LeaseSettings
+
+import scala.concurrent.Future
+
+@ApiMayChange
+abstract class Lease(val settings: LeaseSettings) {
+
+  /**
+   * Try to acquire the lease. The returned `Future` will be completed with `true`
+   * if the lease could be acquired, i.e. no other owner is holding the lease.
+   *
+   * The returned `Future` will be completed with `false` if the lease for certain couldn't be
+   * acquired, e.g. because some other owner is holding it. It's completed with [[akka.lease.LeaseException]]
+   * failure if it might not have been able to acquire the lease, e.g. communication timeout
+   * with the lease resource.
+   *
+   * The lease will be held by the [[akka.lease.LeaseSettings.ownerName]] until it is released
+   * with [[Lease.release]]. A Lease implementation will typically also lose the ownership
+   * if it can't maintain its authority, e.g. if it crashes or is partitioned from the lease
+   * resource for too long.
+   *
+   * [[Lease.checkLease]] can be used to verify that the owner still has the lease.
+   */
+  def acquire(): Future[Boolean]
+
+  /**
+   * Same as acquire with an additional callback
+   * that is called if the lease is lost. The lease can be lose due to being unable
+   * to communicate with the lease provider.
+   * Implementations should not call leaseLostCallback until after the returned future
+   * has been completed
+   */
+  def acquire(leaseLostCallback: Option[Throwable] ⇒ Unit): Future[Boolean]
```

Comment:
Should the param be a specific type rather than `Option[Throwable]` (and have some description string or something in the `None` case)?

Reply:
The other APIs (check lease and acquire lease) just return false as there isn't much information apart from someone else has the lease